framework/src/main/java/org/checkerframework/common/aliasing/AliasingVisitor.java

```diff
@@ -271,7 +265,37 @@ private boolean canBeLeaked(Tree exp) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
         boolean isMethodInvocation = exp.getKind() == Kind.METHOD_INVOCATION;
         boolean isNewClass = exp.getKind() == Kind.NEW_CLASS;
-        return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        boolean isUniqueClassFlag = isUniqueClass(exp);
+        if (isUniqueClassFlag) {
+            return type.hasAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
```

Comment:
Why do you need the `type.hasAnnotation` again, if you already know that `isUniqueClass`?

Reply:
Yes, I see that was a redundant check. I have not used it in my latest commits